src/compat/db_json.hpp

```diff
@@ -80,7 +80,7 @@ void db_json_copy_doc (JSON_DOC *dest, const JSON_DOC *src);
 
 int db_json_insert_func (const JSON_DOC *value, JSON_DOC *doc, char *raw_path);
 int db_json_remove_func (JSON_DOC *doc, char *raw_path);
-int db_json_merge_func (const JSON_DOC *source, JSON_DOC *dest);
+int db_json_merge_func (const JSON_DOC *source, JSON_DOC *&dest);
 
 void db_json_merge_two_json_objects (JSON_DOC *obj1, const JSON_DOC *obj2);
```

Comment:
make db_json_merge_two_json_objects, db_json_merge_two_json_arrays and db_json_merge_two_json_by_array_wrapping static, and change their first argument to JSON_DOC&